src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -3269,18 +3269,9 @@ private void EatPreamble()
             Debug.Assert(_ps.encoding != null);
             Debug.Assert(_ps.bytes != null);
             ReadOnlySpan<byte> preamble = _ps.encoding.Preamble;
-            int preambleLen = preamble.Length;
-            int i;
-            for (i = 0; i < preambleLen && i < _ps.bytesUsed; i++)
-            {
-                if (_ps.bytes[i] != preamble[i])
-                {
-                    break;
-                }
-            }
-            if (i == preambleLen)
+            if (preamble.CommonPrefixLength(_ps.bytes.AsSpan(0, _ps.bytesUsed)) == preamble.Length)
```

Comment:
big thanks for using it here and in other places! 👍  (btw I have no idea how you were able to find this particular code)